python/tests/integration/toolbox/test_library_tool.py

```diff
@@ -230,3 +232,85 @@ def iterate_through_version_chain(key):
     assert len(keys_by_key_type[KeyType.TABLE_DATA]) == (num_versions-1) % 3 + 1
     assert len(keys_by_key_type[KeyType.TOMBSTONE_ALL]) == num_versions // 3
 
+
+def test_overwrite_append_data(object_and_mem_and_lmdb_version_store):
+    lib = object_and_mem_and_lmdb_version_store
+    if lib._lib_cfg.lib_desc.version.encoding_version == 1:
+        # TODO: Fix the timeseries descriptor packing. Currently the [incomplete_segment_from_frame] function in cpp is
+        # not encoding aware so all incomplete writes are broken with v2 encoding.
+        pytest.xfail("Writing the timeseries descriptor for incompletes is currently broken with encoding v2")
+    lib_tool = lib.library_tool()
+    sym = "sym"
+
+    def get_df(num_rows, start_index, col_type):
+        start_date = pd.Timestamp(2024, 1, 1) + pd.Timedelta(start_index, unit="d")
+        index = pd.date_range(start_date, periods=num_rows)
+        df = pd.DataFrame({"col": range(start_index, num_rows+start_index) , "other": range(start_index, num_rows+start_index)}, index=index)
+        # Streaming data has a named index
+        df.index.name = "time"
+        return df.astype({"col": col_type})
+
+    # Deliberately write mismatching incomplete types
+    lib.write(sym, get_df(3, 0, np.int64))
+    lib.write(sym, get_df(1, 3, np.int64), incomplete=True)
+    lib.write(sym, get_df(1, 4, str), incomplete=True)
+    lib.write(sym, get_df(1, 5, np.int64), incomplete=True)
+
+    def read_append_data_keys_from_ref(symbol):
```

Comment:
These helper functions could be added to the library_tool api but I decided against it because it would require frequent tweaks (e.g. if you want to load only the last 20). So instead after this is merged I'll include some examples in the lib tool docs.